### PR TITLE
Remove hardcoded timeout from proxy readiness probe

### PR DESCRIPTION
**Pull request by miguelvr**
checklist:

- [x] run `make test` and `make lint`
- [ ] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

**Review by deliahu — APPROVED** [review]

File: cmd/proxy/main.go [hunk 1]
```diff
@@ -203,10 +204,11 @@ func exit(log *zap.SugaredLogger, err error, wrapStrs ...string) {
 
 func readinessTCPHandler(port int, logger *zap.SugaredLogger) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		timeout := time.Duration(1) * time.Second
-		address := net.JoinHostPort("localhost", strconv.FormatInt(int64(port), 10))
+		ctx := r.Context()
```
Comment (RobertLucian) [inline cmd/proxy/main.go:207]: Shouldn't we use the request's context instead?

Comment (RobertLucian) [reply cmd/proxy/main.go:207]: Oh it's right there, lol.

**Review by RobertLucian — APPROVED** [review]

File: cmd/proxy/main.go (hunk 1 above)
Comment (miguelvr) [reply cmd/proxy/main.go:207]: that's exactly what this is

Comment (RobertLucian) [reply cmd/proxy/main.go:207]: Yep. Thought an empty context was created.

**Pull request merged**